### PR TITLE
feat:  add status for incidents

### LINKED ISSUE #1887: [➕ Feature]: Add status for incidents
Currently, incidents do not have any status. We need to introduce them with the possibility of changing them through the UI. When an incident is marked as resolved, all assigned alerts should be resolved as well.

Statuses:

* firing
* acknowledged
* resolved

### PR DESCRIPTION
**Pull request by VladimirFilonov**
Closes #1887 

## 📑 Description

https://github.com/user-attachments/assets/0df32488-bc73-4f09-8796-978c4aa3510d

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

### CODE REVIEW

**Comment by shahargl** [comment]
@VladimirFilonov 

wdyt expanding the width of the table to whole page and moving the summary above it?
CleanShot 2024-09-14 at 16 09 10@2x

File: keep/api/routes/incidents.py [hunk 1]
```diff
@@ -443,3 +447,49 @@ def confirm_incident(
...
+    tenant_id = authenticated_entity.tenant_id
+    logger.info(
+        "Fetching incident",
+        extra={
+            "incident_id": incident_id,
+            "tenant_id": tenant_id,
+        },
+    )
+
+    incident = get_incident_by_id(tenant_id, incident_id, with_alerts=True)
```
Comment (VladimirFilonov) [inline keep/api/routes/incidents.py:472]:
```suggestion
    with_alerts = change.status == IncidentStatus.RESOLVED
    incident = get_incident_by_id(tenant_id, incident_id, with_alerts=with_alerts)
```

**Review by talboren — APPROVED** [review]
LGTM!

**Review by talboren — APPROVED** [review]
LGTM

**Comment by talboren** [comment]
> @VladimirFilonov
> 
> wdyt expanding the width of the table to whole page and moving the summary above it? CleanShot 2024-09-14 at 16 09 10@2x

Talked with @VladimirFilonov about it and it'll be handled in a different task.

**Review by talboren — APPROVED** [review]
LGTM

**Pull request merged**